src/doc/tarpl/drop-flags.md

```diff
@@ -0,0 +1,81 @@
+% Drop Flags
+
+The examples in the previous section introduce an interesting problem for Rust.
+We have seen that's possible to conditionally initialize, deinitialize, and
+*reinitialize* locations of memory totally safely. For Copy types, this isn't
+particularly notable since they're just a random pile of bits. However types
+with destructors are a different story: Rust needs to know whether to call a
+destructor whenever a variable is assigned to, or a variable goes out of scope.
+How can it do this with conditional initialization?
+
+It turns out that Rust actually tracks whether a type should be dropped or not
+*at runtime*. As a variable becomes initialized and uninitialized, a *drop flag*
+for that variable is toggled. When a variable *might* need to be dropped, this
+flag is evaluated to determine if it *should* be dropped.
+
+Of course, it is *often* the case that a value's initialization state can be
+*statically* known at every point in the program. If this is the case, then the
+compiler can theoretically generate more effecient code! For instance, straight-
+line code has such *static drop semantics*:
+
+```rust
+let mut x = Box::new(0); // x was uninit; just overwrite.
+let mut y = x;           // y was uninit; just overwrite and make x uninit.
+x = Box::new(0);         // x was uninit; just overwrite.
+y = x;                   // y was init; Drop y, overwrite it, and make x uninit!
+                         // y was init; Drop y!
+                         // x was uninit; do nothing.
+```
```

Comment:
nit: if it were me, I'd (1.) include explicit block-curly braces `{` and `}` and then attach the drop notes to the close braces, maybe even put each let binding in distinct nested blocks, and (2.) include a case of a uninitialized let-binding above; this will soften the blow when you use it in your example below.

(But these are relatively minor issues.)